app/code/FrontCommerce/ChronorelaisHeadless/Model/RelayPointManagement.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace FrontCommerce\ChronorelaisHeadless\Model;
+
+use Exception;
+use FrontCommerce\ChronorelaisHeadless\Api\Data\RelayPointsResponseInterface;
+use FrontCommerce\ChronorelaisHeadless\Api\RelayPointManagementInterface;
+use Magento\Checkout\Model\Session;
+use Magento\Quote\Api\CartRepositoryInterface;
+
+class RelayPointManagement implements RelayPointManagementInterface
+{
+    protected const METHOD_CODE = 'chronorelais';
+
+    protected RelayPoint $relayPoint;
+    protected RelayPointsResponseInterface $relayPointsResponse;
+    protected Session $session;
+    protected CartRepositoryInterface $cartRepository;
+
+    public function __construct(
+        RelayPoint                   $relayPoint,
+        RelayPointsResponseInterface $relayPointsResponse,
+        Session                      $session,
+        CartRepositoryInterface      $cartRepository
+    )
+    {
+        $this->relayPoint = $relayPoint;
+        $this->relayPointsResponse = $relayPointsResponse;
+        $this->cartRepository = $cartRepository;
+        $this->session = $session;
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function getRelayPoints(string $postCode, array $shippingAddress): array
+    {
+        $this->checkParams($postCode, $shippingAddress);
+
+        return $this->formatRelayResponseData(
+            $this->relayPoint->getRelayPoints(self::METHOD_CODE, $postCode, $shippingAddress)
+        );
+    }
+
+    protected function formatRelayResponseData(array $relayPoints): array
+    {
+        if (!$relayPoints) {
+            return [];
+        }
+
+        $result = [];
+        $relayPoints = json_decode(json_encode($relayPoints));
+
+        foreach ($relayPoints as $relayPoint) {
+            $relayPoint = json_decode(json_encode($relayPoint));
+
+            $result[] = clone $this->relayPointsResponse
+                ->setStreet1($relayPoint->adresse1)
+                ->setStreet2($relayPoint->adresse2)
+                ->setStreet3($relayPoint->adresse3)
+                ->setLatitude($relayPoint->latitude)
+                ->setLongitude($relayPoint->longitude)
+                ->setZipCode($relayPoint->codePostal)
+                ->setRelayId($relayPoint->identifiantChronopostPointA2PAS)
+                ->setName($relayPoint->nomEnseigne)
+                ->setCity($relayPoint->localite)
+                ->setName($relayPoint->nomEnseigne)
+                ->setOpeningTime([
+                    'Monday' => $relayPoint->horairesOuvertureLundi,
+                    'Tuesday' => $relayPoint->horairesOuvertureMardi,
+                    'Wednesday' => $relayPoint->horairesOuvertureMercredi,
+                    'Thursday' => $relayPoint->horairesOuvertureJeudi,
+                    'Friday' => $relayPoint->horairesOuvertureVendredi,
+                    'Saturday' => $relayPoint->horairesOuvertureSamedi,
+                    'Sunday' => $relayPoint->horairesOuvertureDimanche,
+                ]);
```

Comment:
this results in:

```json
{
  "relay_id": "6055S",
  "street1": "6 rue Saint Louis",
  "street2": "",
  "street3": "",
  "latitude": "43.61611111110",
  "longitude": "3.15583333333",
  "zip_code": "34600",
  "city": "BEDARIEUX",
  "name": "TABAC LOTO DU PONT D ORB",
  "opening_time": [
    "07:00-12:00 12:00-19:30",
    "07:00-12:00 12:00-19:00",
    "07:00-12:00 12:00-19:00",
    "07:00-12:00 12:00-19:00",
    "07:00-12:00 12:00-19:00",
    "07:00-12:00 12:00-19:00",
    "00:00-00:00 00:00-00:00"
  ]
}
```

while the expected format is:

```json
{
    "street1": "64 Rue principale",
    "street2": "",
    "street3": "",
    "latitude": "46.13221716813",
    "longitude": "5.16451953636",
    "zipcode": "01960",
    "id": "2039V",
    "city": "Servas",
    "name": "Tabac au rendez vous des amis",
    "openingTime": {
        "monday": "07:00-12:00 12:00-18:30",
        "tuesday": "07:00-12:00 12:00-18:30",
        "wednesday": "07:00-12:00 12:00-18:30",
        "thursday": "07:00-12:00 12:00-18:30",
        "friday": "07:00-12:00 12:00-18:30",
        "saturday": "07:00-12:00 12:00-18:30",
        "sunday": "07:00-12:00 12:00-13:00"
    }
}
```

the camelcase vs. snakecase and the id vs. relay_id changes are not a big issue but not having the days in opening time is an issue.

Reply:
openingTime is still not correct:

```json
{
    "street1": "6 rue Saint Louis",
    "street2": "",
    "street3": "",
    "latitude": "43.61611111110",
    "longitude": "3.15583333333",
    "zipcode": "34600",
    "id": "6055S",
    "city": "BEDARIEUX",
    "name": "TABAC LOTO DU PONT D ORB",
    "openingTime": [
      "07:00-12:00 12:00-19:30",
      "07:00-12:00 12:00-19:00",
      "07:00-12:00 12:00-19:00",
      "07:00-12:00 12:00-19:00",
      "07:00-12:00 12:00-19:00",
      "07:00-12:00 12:00-19:00",
      "00:00-00:00 00:00-00:00"
    ]
  }
```

the days are missing